web/modules/custom/joinup_notification/joinup_notification.behat.inc

```diff
@@ -70,11 +70,11 @@ class JoinupNotificationSubContext extends DrupalSubContextBase implements Drupa
     $html = empty($html) || $html === 'yes';
 
     $user = user_load_by_name($recipient);
-    if (empty($user)) {
-      throw new \Exception("User $recipient was not found.");
+    if (!empty($user)) {
+      $recipient = $user->getEmail();
     }
-    if (empty($user->getEmail())) {
-      throw new \Exception("User $recipient does not have an e-mail address.");
+    elseif (!filter_var($recipient, FILTER_VALIDATE_EMAIL)) {
+      throw new \Exception("Recipient with user/mail {$user} could not be understood.");
```

Comment:
Shouldn't this be `Recipient {$recipient} could not be understood.`?

`$user` is empty at this point.